src/3d/qgstessellatedpolygongeometry.h

```diff
@@ -108,6 +108,18 @@ class QgsTessellatedPolygonGeometry : public Qt3DCore::QGeometry
      */
     QgsFeatureId triangleIndexToFeatureId( uint triangleIndex ) const;
 
+    //! Returns included feature ids
+    QVector<QgsFeatureId> featureIds() const { return mTriangleIndexFids; }
+
+    //! Returns triangle index for features. For a feature featureIds()[i], matching triangles start at triangleIndexStartingIndices()[i]
+    QVector<uint> triangleIndexStartingIndices() const { return mTriangleIndexStartingIndices; }
+
+    /**
+     * Tests whether the geometry is intersected by \a ray.
+     * In case of success, the \a intersectionPoint and the corresponding \a fid are updated.
+     */
+    bool rayIntersection( const QgsRayCastingUtils::Ray3D &ray, const QMatrix4x4 &worldTransform, QVector3D &intersectionPoint, QgsFeatureId &fid );
```

Comment:
I think this is a result of a bad merge conflict fix -- it shouldn't be re added here

Reply:
indeed!